src/map.cpp

```diff
@@ -154,6 +154,44 @@ void Map::setTile(uint16_t x, uint16_t y, uint8_t z, Tile* newTile)
 	}
 }
 
+void Map::removeTile(uint16_t x, uint16_t y, uint8_t z)
+{
+	if (z >= MAP_MAX_LAYERS) {
+		return;
+	}
+
+	const QTreeLeafNode* leaf = QTreeNode::getLeafStatic<const QTreeLeafNode*, const QTreeNode*>(&root, x, y);
+	if (!leaf) {
+		return;
+	}
+
+	const Floor* floor = leaf->getFloor(z);
+	if (!floor) {
+		return;
+	}
+
+	Tile* tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
+	if (tile) {
+		if (const CreatureVector* creatures = tile->getCreatures()) {
+			for (int32_t i = creatures->size(); --i >= 0;) {
+				g_game.removeCreature((*creatures)[i]);
```

Comment:
Tested the code and works as it should, but not sure about this part.
What happens now is you can still see the ingame screen but your char is removed (just like when you use /r on a player)
Might be a good ide to kick the player and maybe even teleport it to temple, just tested to remove all the tiles around my char and got a client debug when I tried to log back in.

Reply:
I like the idea of teleporting them to the temple in this odd case